docs/templates.rst

```diff
@@ -640,9 +640,12 @@ iterate over containers like `dict`::
     {% endfor %}
     </dl>
 
-Note, however, that **Python dicts are not ordered**; so you might want to
-either pass a sorted ``list`` of ``tuple`` s -- or a
-``collections.OrderedDict`` -- to the template, or use the `dictsort` filter.
+Note, however, that **Python dicts preserve insertion order**, which may not
+be the order you expect. If order matters, you can sort data.items() or use
+the dictsort filter. (In Python versions prior to 3.7, ``dict`` objects are
+unsorted, so you might want to either pass a sorted ``list`` of ``tuple`` s
+-- or a ``collections.OrderedDict`` -- to the template, or use the `dictsort`
+filter.)
```

Comment:
I actually like splitting this thematically into sorting before passing the dict to the template or sorting it inside the template using jinja filters like you did in the second sentence. It just feels like these could be fused into one with a bit rewording. (You could still pass a sorted list of tuples and you could also sort the items before 3.7)

Reply:
You may be right that the wording could be improved, but I think in either case this is still better than the existing documentation, which is still displaying misleading information 5 months later. I guess this is just a polite reminder that this exists and could use some love.